extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -598,11 +598,16 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
 
                     if (_enableAjaxPerfTracking) {
                         let iKey = config.instrumentationKey || "unkwn";
-                        if (iKey.length > 5) {
-                            _markPrefix = AJAX_MONITOR_PREFIX + strSubstring(iKey, iKey.length - 5) + ".";
-                        } else {
-                            _markPrefix = AJAX_MONITOR_PREFIX + iKey + ".";
+                        // only change the ikey if it is string
```

Comment:
Lets drop this code (for now) as this is the bit that Siyu is now looking at. If you need to make it compile just cast the instrumentationKey to a string (to minimize the amount of merge conflicts this is going to cause).